chia/wallet/nft_wallet/uncurry_nft.py

```diff
@@ -0,0 +1,108 @@
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+
+from chia.types.blockchain_format.program import Program
+from chia.wallet.puzzles.load_clvm import load_clvm
+
+SINGLETON_TOP_LAYER_MOD = load_clvm("singleton_top_layer_v1_1.clvm")
+NFT_MOD = load_clvm("nft_innerpuz.clvm")
+
+
+@dataclass
+class UncurriedNFT:
+    """
+    A simple solution for uncurry NFT puzzle.
+    Initial the class with a full NFT puzzle, it will do a deep uncurry.
+    This is the only place you need to change after modified the Chialisp curried parameters.
+    """
+
+    matched: bool = field(init=False)
+    """If the puzzle is a NFT puzzle"""
+
+    nft_mod_hash: Program = field(init=False)
+    """NFT module hash"""
+
+    singleton_struct: Program = field(init=False)
+    """
+    Singleton struct
+    [singleton_mod_hash, singleton_launcher_id, launcher_puzhash]
+    """
+    singleton_mod_hash: Program = field(init=False)
+    singleton_launcher_id: Program = field(init=False)
+    launcher_puzhash: Program = field(init=False)
+
+    owner_did: Program = field(init=False)
+    """Owner's DID"""
+
+    transfer_program_hash: Program = field(init=False)
+    """Puzzle hash of the transfer program"""
+
+    transfer_program_curry_params: Program = field(init=False)
+    """
+    Curried parameters of the transfer program
+    [royalty_address, trade_price_percentage, settlement_mod_hash, cat_mod_hash]
+    """
+    royalty_address: Program = field(init=False)
+    trade_price_percentage: Program = field(init=False)
+    settlement_mod_hash: Program = field(init=False)
+    cat_mod_hash: Program = field(init=False)
+
+    metadata: Program = field(init=False)
+    """
+    NFT metadata
+    [("u", data_uris), ("h", data_hash)]
+    """
+    data_uris: Program = field(init=False)
+    data_hash: Program = field(init=False)
+
+    @staticmethod
```

Comment:
`@classmethod` not `@staticmethod`, use the `cls` parameter, and generally figure out what a class is going to be before building it instead of building entirely the wrong instance and then modifying everything.

Reply:
Good to know, if you will work on this I will leave it to you.

Reply:
Being worked over in https://github.com/Chia-Network/chia-blockchain/pull/11379.  I'll ping you for a review when it is ready.